common/database_handler.py

```diff
@@ -71,5 +81,8 @@ def __init__(self):
     def __enter__(self):
         return self.conn
 
-    def __exit__(self, *_):
+    def __exit__(self, exception_type, exception_value, traceback):
+        if exception_type is not None and exception_value is not None and traceback is not None:
```

Comment:
I think it's enough to check only if `exception_type` is not `None`, not necessary to check all 3 attributes

additionaly, I'm currious if it's possible to check if `exception_type` is some `psycopg2` DB connection error to not recover connection when some other exception occurs

and it'd be nice to put an error message with value a traceback to log if it's not done on some other place already (I'm not sure)